storage/remote/queue_manager.go

```diff
@@ -932,6 +997,24 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue chan sample) {
 			s.droppedOnHardShutdown.Add(uint32(droppedSamples))
 			return
 
+		case <-s.reshard:
+			var samples []sample
+			for {
+				sample, ok := <-queue
+				// Accept those samples that might have entered the queue between sending the resharding
+				// signal and blocking of incoming samples at the time of creation of newer queues.
+				if !ok {
+					break
+				}
+				samples = append(samples, sample)
+			}
+			if samplesCount := len(samples); samplesCount > 0 {
+				level.Info(s.qm.logger).Log("msg", fmt.Sprintf("Sending %d samples to newer queues.", samplesCount))
```

Comment:
This could be very noisy when many shards are being run, perhaps it would be better to move this to debug level and have the overall `bufferize` function print a line at info level about the total # of samples passed to the new shards?